frontend/src/pages/NewRun.test.tsx

```diff
@@ -783,7 +769,103 @@ describe('NewRun', () => {
         mode: 'error',
       }));
     });
+  });
+
+  describe('arriving from pipeline details page', () => {
+
+    it('indicates that a pipeline is preselected and provides a means of selecting a different pipeline', async () => {
+      const props = generateProps();
+      props.location.search = `?${QUERY_PARAMS.fromRunId}=${MOCK_RUN_WITH_EMBEDDED_PIPELINE.run!.id}`;
```

Comment:
It would be nice if these two lines move to the `describe` level, make the code more readable IMO.